docs/pages/pmd/userdocs/suppressing_warnings.md

```diff
@@ -187,8 +187,13 @@ of no use.
 ###  The property `violationSuppressXPath`
 
 This property defines an XPath query to be executed *using the
-violation node as the starting point*.  If the XPath query matches anything,
-then the violation will be suppressed.
+violation node as the context node*.  If the XPath query matches anything,
+then the violation will be suppressed. Note that the query shouldn't be finding
+the violation nodes to suppress, but rather, finding a non-empty sequence of nodes
+when evaluated with the violation node as a context node.
+
+The XPath version used by those queries is XPath 1.0, so it doesn't support e.g.
+regex tests. This will be updated with PMD 7.0.0.
```

Comment:
Actually we do support regex - we have a custom function called `matches`: https://github.com/pmd/pmd/blob/master/pmd-core/src/main/java/net/sourceforge/pmd/lang/xpath/MatchesFunction.java

```
./ancestor::ClassOrInterfaceDeclaration[matches(@Image, 'Bean')]
```

In XPath 2.0, this function is part of XPath already.

The problem is, I guess, that we don't have a documentation about the various XPath functions, that are available in PMD... (java also provides own functions like typeIs)

Reply:
I've update the sentence and added an additional example with matches.